### PR TITLE
:sparkles: Adding Logger that enforces log level

### PR DESCRIPTION
**Pull request by shawn-hurley**
Disallow for an individual component to change the level of the log for the program; instead, force the level set in the configuration file to all the logs.

I will complete the tracing after this merges and after #451 is merged.

### CODE REVIEW

File: kai/logging/kai_trace.py
Comment (JonahSussman) [inline kai/logging/kai_trace.py:1]: Not a huge fan of how KaiTrace was implemented in the past. We should probably think of a better way to do tracing later.

File: kai/logging/logging.py [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+
+
+class KaiLogger(logging.Logger):
+    configLogLevel: int
+
+    def __init__(self, name: str, configLogLevel: int) -> None:
+        super().__init__(name, configLogLevel)
+        self.configLogLevel = configLogLevel
+
+    @override
```
Comment (JonahSussman) [inline kai/logging/logging.py:17]: Not sure if `@override` is necessary

File: playpen/old_rpc_stuff/client/rpc.py
Comment (JonahSussman) [inline playpen/old_rpc_stuff/client/rpc.py:1]: I think we can probably remove `old_rpc_stuff`, no?

File: tests/test_kai_logging.py [hunk 2]
```diff
@@ -77,30 +77,9 @@ def test_init_logging(self):
...
-        mock_getcwd.return_value = base_dir
-        expected_log_dir = os.path.join(base_dir, "logs")
-        init_logging_from_config(self.config)
-
-        mock_makedirs.assert_called_once_with(expected_log_dir, exist_ok=True)
-
-        logger = logging.getLogger("kai")
+        logger = get_logger("test")
         self.assertEqual(logger.level, logging.DEBUG)
+        print(f"herelrhelajdlfkjadlkjfsdlfjkasdlfkj {logger.handlers}")
```
Comment (JonahSussman) [inline tests/test_kai_logging.py:82]: lol

**Review by JonahSussman — APPROVED** [review]
A few minor things

**Review by pranavgaikwad — APPROVED** [review]

**Pull request merged**